h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -57,6 +59,42 @@ public boolean isSupervised() {
     return true;
   }
 
+  @Override
+  protected void ignoreBadColumns(int npredictors, boolean expensive){
+    String msg = "Dropping bad and constant columns: ";
+
+    HashSet usedColumns = new HashSet();
+
+    for(Key k: _parms._base_models) {
+      Model model = (Model) DKV.getGet(k);
+      usedColumns.add(model._parms._response_column);
+      usedColumns.addAll(Arrays.asList(model._parms.getNonPredictors()));
+      if (model._output._origNames != null)
+        usedColumns.addAll(Arrays.asList(model._output._origNames));
+      else
+        usedColumns.addAll(Arrays.asList(model._output._names));
+    }
+
+    usedColumns.addAll(Arrays.asList(_parms.getNonPredictors()));
+
+    if (usedColumns.size() < _train._names.length) {
```

Comment:
looks like this condition makes too strong assumptions: we don't require all base models to have been trained with the same training frame (I mean using the same predictors). The only requirement somehow, is that the training frame used to train the SE should have enough columns to allow each model to give predictions, knowing that models don't usually need all their training predictors to produce predictions.

This being said, we can imagine cases were `usedColumns > train.length` but you could still be able to remove columns from train, so I would just remove the condition.

Reply:
This is very good point. I completely missed that we can use different training frames. Thanks!